platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -289,26 +264,25 @@ public int delete(String type, String cql) throws PersistenceException {
         return idsToDelete.size();
     }
 
-    private SolrClient getSolrCore(String storeName) {
-        if (coreSolrClients.containsKey(storeName)) {
-            LOGGER.debug("Returning core {} from map of coreSolrClients", storeName);
-            return coreSolrClients.get(storeName);
+    private SolrClient getSolrClient(String storeName) {
```

Comment:
The solrClients map is a ConcurrentHashMap.  The race was not a concern since the worst case was the racing threads would both get a client and set the map.  I am not sure if the cost of building the extra client is worth the extra complexity to synchronize.  Is there a different concurrency issue I am missing?